src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,289 @@
+        Spark.init();
+
+        Spark.get("/",
+                ((request, response) -> {
+                    HashMap m = new HashMap();
+
+                    //This is the new sessions login that uses the database instead of the array
+                    Session session = request.session();
+                    String name = session.attribute("userName");
+                    if (name == null) {
+                        return new ModelAndView(m, "login.html");
+                    }
+                    User user = selectUser(conn,name);
+                    ArrayList<Messages> msgs = selectAllMessages(conn,user.getUserName());
+                    m.put("messages", msgs);
+                    m.put("userName", user.getUserName());
+                    return new ModelAndView(m, "messages.html");
+
+                }),
+                new MustacheTemplateEngine()
+        );
+
+        Spark.post("/login",
+                ((request, response) -> {
+                    String loginName = request.queryParams("loginName");
+                    String password = request.queryParams("password");
+
+//                    boolean checkValid = users.containsKey(loginName);
+//                    Session session = request.session();
+//                    String name = session.attribute("userName");
+//                    User user = users.get(name);
+//                    //this catches an empty username box to prompt them into the create menu
+//                    //the next few blocks are meant as catchers for the incorrect login scenario
+//                    if (user==null) {
+//                        response.redirect("/login");
+//                    }
+                    if(loginName==null||password==null){
+                        response.redirect("/");
+                        return"";
+                    }
+                    if(!userNameInUse(conn,loginName)){
+                        //send boolean for already in use
+                        //HTML TRY ANOTHER NAME
+                        response.redirect("/");
+                        return "";
+                    }
+                    User user=selectUser(conn,loginName);
+
+                    if(user.getUserName().equalsIgnoreCase(loginName) && user.getPassword().equals(password)) {
+
+                        Session session = request.session();
+                        session.attribute("userName", loginName);
+                        response.redirect("/");
+                        return "";
+                    }
+                    response.redirect("/login");
+                    return "";
+                })
+        );
+
+        Spark.post("/delete-entry",
+                (((request, response) -> {
+                    int someId = Integer.valueOf(request.queryParams("postId")) ;
+                    deleteMessage(conn,Integer.valueOf(request.queryParams("postId")));
+                    response.redirect("/");
```

Comment:
you are actually bringing in the value of messages.postID with this queryParams so you need to change your delete SQL statement to reflect that.